src/components/button/index.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import styles from './button.scss';
+import classNames from 'classnames';
+/**
+ *
+ * @param {children} props
+ * @param {type} props  primary, danger, secondary
+ * @param {size} props  default, large, small,
+ * @param {disabled} props  true, false
+ * @param {fullWidth} props Button width is 100% of parent
+ */
+const Button = (props) => {
+  const { children, type, size, disabled, fullWidth, ...rest } = props;
+  const classes = classNames(
+    styles.button,
+    styles[type] || styles.primary,
+    styles[size] || styles.default,
+    fullWidth && styles.fullWidth,
+  );
+  return (
+    <button disabled={disabled} {...rest} className={classes}>
```

Comment:
I'm not sure I'm following you xD but yes, ref forwarding sound like it might be useful at some point

Reply:
im saying something like this

```javascript
handleClick = (event) = > {
   const {currentTarget} = event;
   const someId = currentTarget.getAttribute('id')
   // useSomeId for further calls
}

${users.map(user => <Button id={user.id} onClick={this.handleClick}> View Details</Button>)}
```

In this case i can use the id props and the same handler. Advantage of this is i wont have to use arrow function, so on each render a new function isnt created

Reply:
Sorry my ignorance... but I still don't see the point of this. :pray:

Reply:
Umm okay maybe let me explain it a bit better. Say you have an array of items like so
```javascript
items = [
  { id : 1, type: 'bread', price: 200},
  { id : 1, type: 'milk', price: 100},
  { id : 1, type: 'butter', price: 50}
]
```
And i want to remove an item from the array, you would do something like this

```javascript
handleRemove(item) {
 // remove logic here
}

items.map(item => <button onClick={(item) => handleRemove(item)}>Remove</button>
```
The `slight` downside of this is every time i have a state change a new function for the onclick handler is created. 
So what i was thinking was we can use the `event` object which is passed into every event handler and use some data which is an attribute on the element to handle the same logic.
For example
Consider the same case as above
```javascript
handleRemove = (event) => {
  const {currentTarget} = event;
 const id = currentTarget.getAttributes('id')
 // now handle remove logic
}

items.map(item => <button onClick={this.handleRemove} id={item.id}>Remove</button>
```

Maybe im overthinking, but i feel this can be useful sometime later.. anyway if it seems not so useful wont add it

Reply:
Oh okay! I get it now! Thank you for the exaplanation! :smile:  I like the idea! The only this I don't like is the fact to add a prop to a component only for this kind of uses.

Like, of course it's good to be able to give an `id` to a `<button>` but, would we give it to him if we did not need the `id` for exactly this case of optimization? I mean, I am down I am not going to say no, just trying how this impact! ^^

For me it's a yes though let's wait the others

Reply:
This is an interesting idea, I'd prefer to define a `data` property (for example) that receives an object with all the data you want to add to the dom, then use the **data-**something-here attributes for each property.

```
<Button data={{ id: 1 }} />

// renders to the dom: <button data-id="1">
```

The you can access the data in the event like this:
```
event.currentTarget.dataset
```

Anyway... this is not the react way of doing things, so.... While I'm open for something like this, I'm still a little hesitant xD